internal/driver/device.go

```diff
@@ -91,16 +91,16 @@ func (dev *Device) StopStreaming() {
 	}
 }
 
-// SetFps updates the fps on the device side of the service. Note that this won't update the rtsp output stream fps
-func (dev *Device) SetFrameRate(usbdevice *usbdevice.Device, fpsNumerator uint32, fpsDenominator uint32) (string, error) {
-	fps := fmt.Sprintf("%f", float32(fpsDenominator)/float32(fpsNumerator))
-	dataFormat, err := getDataFormat(usbdevice)
+// SetFrameRate updates the fps on the device side of the service. Note that this won't update the rtsp output stream fps
+func (dev *Device) SetFrameRate(usbDevice *usbdevice.Device, intervalNumerator uint32, intervalDenominator uint32) (string, error) {
```

Comment:
i would switch this function to just taking the fps values as is, and comparing them as is, and then just invert when setting the timePerFrame.